test/integration/node/lifecycle_test.go

```diff
@@ -248,123 +268,143 @@ func TestTaintBasedEvictions(t *testing.T) {
 		defaulttolerationseconds.NewDefaultTolerationSeconds(),
 	)
 	for _, test := range tests {
-		t.Run(test.name, func(t *testing.T) {
-			testCtx := testutils.InitTestAPIServer(t, "taint-based-evictions", admission)
+		for _, enableSeparateTaintManager := range []bool{true, false} {
```

Comment:
per previous comment, it's more conventional to test one scenario per sub-test, so I'd suggest to add extra sub-tests to cover enableSeparateTaintManager=false case, and here we don't iterate it.

Reply:
Changed this in 21859779c3c. I took this approach initially to avoid having to duplicate the test-cases but we can totally change it to follow the same approach as other tests.